### PR TITLE
Compact rcf

### PR DESCRIPTION
**Pull request by sudiptoguha**
*Description of changes:* fixes the V1 -> V2 tests and converters. Also cleans up some of the projection code.

### CODE REVIEW

File: Java/serialization/src/test/java/com/amazon/randomcutforest/serialize/json/v1/V1JsonToV2StateConverterTest.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+                b.append(line);
+            }
+
+            String json = b.toString();
+            RandomCutForestState state = converter.convert(json);
+
+            assertEquals(jsonResource.getDimensions(), state.getDimensions());
+            assertEquals(jsonResource.getNumberOfTrees(), state.getNumberOfTrees());
+            assertEquals(jsonResource.getSampleSize(), state.getSampleSize());
+            state.setNumberOfTrees(1);
```
Comment (jotok) [inline Java/serialization/src/test/java/com/amazon/randomcutforest/serialize/json/v1/V1JsonToV2StateConverterTest.java:64]: This was probably for testing, right? Can we remove it?

File: Java/serialization/src/test/java/com/amazon/randomcutforest/serialize/json/v1/V1JsonToV2StateConverterTest.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            RandomCutForest forest = new RandomCutForestMapper().toModel(state,0);
+
+            assertEquals(jsonResource.getDimensions(), forest.getDimensions());
+           // assertEquals(jsonResource.getNumberOfTrees(), forest.getNumberOfTrees());
```
Comment (jotok) [inline Java/serialization/src/test/java/com/amazon/randomcutforest/serialize/json/v1/V1JsonToV2StateConverterTest.java:68]: Uncomment assertion.

File: Java/serialization/src/test/java/com/amazon/randomcutforest/serialize/json/v1/V1JsonToV2StateConverterTest.java (hunk 1 above)
Comment (sudiptoguha) [reply Java/serialization/src/test/java/com/amazon/randomcutforest/serialize/json/v1/V1JsonToV2StateConverterTest.java]: Indeed :) Done.

File: Java/serialization/src/test/java/com/amazon/randomcutforest/serialize/json/v1/V1JsonToV2StateConverterTest.java (hunk 2 above)
Comment (sudiptoguha) [reply Java/serialization/src/test/java/com/amazon/randomcutforest/serialize/json/v1/V1JsonToV2StateConverterTest.java]: Done.

**Pull request merged**